### PR TITLE
Updated Certstore to correctly understand CurrentUser vs LocalMachine stores

### PR DESCRIPTION
**Pull request by johnmccrae**
Signed-off-by: John McCrae <jmccrae@chef.io>

### Description

[Please describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: spec/win32/unit/certstore_spec.rb [hunk 1]
```diff
@@ -325,8 +261,7 @@
       end
       it "returns nil" do
         store = certstore.open(store_name)
-        cert_obj = store.get(thumbprint)
-        expect(cert_obj).to eql(nil)
+        expect { store.get(thumbprint, store_location: CERT_SYSTEM_STORE_CURRENT_USER, store_name: store_name) }.to raise_error(ArgumentError, "Unable to retrieve the certificate")
```
Comment (marcparadise) [inline spec/win32/unit/certstore_spec.rb:264]: For this and below, we should change the description to accurately talk about what is being tested - something like "raises an ArgumentError".

File: spec/win32/functional/win32/certstore_spec.rb [hunk 2]
```diff
@@ -34,34 +34,15 @@
...
-      file = ::File.basename(out_put.strip)
-      expect(file).to eq(cert_pfx)
-    end
-
-    # passing invalid thumbprint
-    it "returns nil if certificate not found" do
-      thumbprint = "b1bc968bd4f49d622aa89a81f2150152a41d829cab"
-      expect(@store.get_pfx(thumbprint, export_password: "1234")).to eq("")
+      expect(@store).to receive(:cert_get).with(thumbprint, store_location: CERT_SYSTEM_STORE_CURRENT_USER, store_name: "My").and_return(nil)
+      @store.get(thumbprint, store_location: CERT_SYSTEM_STORE_CURRENT_USER, store_name: "My")
```
Comment (marcparadise) [inline spec/win32/functional/win32/certstore_spec.rb:45]: This isn't testing the return , should it be? (based on description)

File: spec/win32/unit/certstore_spec.rb [hunk 3]
```diff
@@ -188,70 +188,6 @@
 
   CERT_SYSTEM_STORE_LOCAL_MACHINE = 0x00020000
 
-  describe "#get_cert_pfx" do
```
Comment (marcparadise) [inline spec/win32/unit/certstore_spec.rb:191]: We removed this, but did not add test for `#get_cert_key` which seems to replace it.

**Review by marcparadise — COMMENTED** [review]
Some questions/comments inline

File: spec/win32/unit/certstore_spec.rb (hunk 1 above)
Comment (johnmccrae) [reply spec/win32/unit/certstore_spec.rb:264]: Oh phooey... good catch

**Review by marcparadise — APPROVED** [review]

**Pull request merged**